connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectFetchErrorDetailsHandler.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.service
+
+import io.grpc.stub.StreamObserver
+
+import org.apache.spark.connect.proto
+import org.apache.spark.connect.proto.FetchErrorDetailsResponse
+import org.apache.spark.sql.connect.config.Connect
+import org.apache.spark.sql.connect.utils.ErrorUtils
+import org.apache.spark.sql.internal.SQLConf
+
+class SparkConnectFetchErrorDetailsHandler(
+    responseObserver: StreamObserver[proto.FetchErrorDetailsResponse]) {
+
+  def handle(v: proto.FetchErrorDetailsRequest): Unit = {
+    val sessionHolder =
+      SparkConnectService
+        .getOrCreateIsolatedSession(v.getUserContext.getUserId, v.getSessionId)
+
+    val response = Option(sessionHolder.errorIdToError.getIfPresent(v.getErrorId))
+      .map { error =>
+        ErrorUtils.throwableToFetchErrorDetailsResponse(
+          st = error,
+          isTruncated = false,
+          serverStackTraceEnabled =
+            sessionHolder.session.conf.get(Connect.CONNECT_SERVER_STACKTRACE_ENABLED),
+          pySparkJVMStackTraceEnabled =
+            sessionHolder.session.conf.get(SQLConf.PYSPARK_JVM_STACKTRACE_ENABLED),
```

Comment:
For me as the reader it's hard to reason about that the difference between the two is?

Reply:
Connect.CONNECT_SERVER_STACKTRACE_ENABLED is a more language agnostic conf for returning stack trace.

SQLConf.PYSPARK_JVM_STACKTRACE_ENABLED is python client specific

Reply:
Why can't you merge this into one flag. You could even have two outside, but why does this need to be two here?

Reply:
If we merge this into one flag, the scala client may be affected by PYSPARK_JVM_STACKTRACE_ENABLED as well. But I think it is probably ok